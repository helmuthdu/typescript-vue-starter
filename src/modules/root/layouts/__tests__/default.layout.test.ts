import { createLocalVue, shallowMount, Wrapper } from '@vue/test-utils';
import VueRouter from 'vue-router';
import Vuetify from 'vuetify';
import DefaultLayout from '../default.layout.vue';

const localVue = createLocalVue();
const router = new VueRouter();

localVue.use(VueRouter);
localVue.use(Vuetify);

describe('Home/Layout -> Default', () => {
  let wrapper: Wrapper<DefaultLayout>;
  let vuetify;

  beforeEach(() => {
    vuetify = new Vuetify();
    wrapper = shallowMount(DefaultLayout, {
      localVue,
      vuetify,
      router
    });
  });

  it('should renders with props', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
