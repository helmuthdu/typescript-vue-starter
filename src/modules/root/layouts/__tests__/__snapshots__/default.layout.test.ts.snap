// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout -> default should renders with props 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "__props": Array [],
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
      id="app"
    >
      <div>
        <section
          class="hero is-medium is-primary"
        >
          <div
            class="hero-body"
          >
            <div
              class="container"
            >
              <div
                class="columns"
              >
                <div
                  class="column is-8-desktop is-offset-2-desktop"
                >
                  <h1
                    class="title is-2 is-spaced"
                  >
                     Hello Bulma! 
                  </h1>
                  <h2
                    class="subtitle is-4"
                  >
                     Congratulations! You're running the 
                    <strong>
                      Bulma start
                    </strong>
                     project. 
                    <br />
                     It includes everything you need to 
                    <strong>
                      build your own website
                    </strong>
                     with Bulma. 
                  </h2>
                </div>
              </div>
            </div>
          </div>
        </section>
        <section
          class="section"
        >
          <div
            class="container"
          >
            <div
              class="columns"
            >
              <div
                class="column is-8-desktop is-offset-2-desktop"
              >
                <div
                  class="content"
                >
                  <h3>
                    What's included
                  </h3>
                  <p>
                    The 
                    <code>
                      npm
                    </code>
                     dependencies included in 
                    <code>
                      package.json
                    </code>
                     are:
                  </p>
                  <ul>
                    <li>
                      <code>
                        <a
                          href="https://github.com/jgthms/bulma"
                        >
                          bulma
                        </a>
                      </code>
                    </li>
                    <li>
                      <code>
                        <a
                          href="https://github.com/sass/node-sass"
                        >
                          node-sass
                        </a>
                      </code>
                       to compile your own Sass file 
                    </li>
                    <li>
                      <code>
                        <a
                          href="https://github.com/postcss/postcss-cli"
                        >
                          postcss-cli
                        </a>
                      </code>
                       and 
                      <code>
                        <a
                          href="https://github.com/postcss/autoprefixer"
                        >
                          autoprefixer
                        </a>
                      </code>
                       to add support for older browsers 
                    </li>
                    <li>
                      <code>
                        <a
                          href="https://babeljs.io/docs/usage/cli/"
                        >
                          babel-cli
                        </a>
                      </code>
                      , 
                      <code>
                        <a
                          href="https://github.com/babel/babel-preset-env"
                        >
                          babel-preset-env
                        </a>
                      </code>
                       and 
                      <code>
                        <a
                          href="https://github.com/jmcriffey/babel-preset-es2015-ie"
                        >
                          babel-preset-es2015-ie
                        </a>
                      </code>
                       for compiling ES6 JavaScript files 
                    </li>
                  </ul>
                  <!---->
                </div>
                <table
                  class="table is-bordered is-fullwidth"
                >
                  <thead>
                    <tr>
                      <th>
                        Type
                      </th>
                      <th>
                        CSS class
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr>
                      <th>
                        Columns
                      </th>
                      <td>
                        <code>
                          <a
                            href="http://bulma.io/documentation/columns/basics"
                          >
                            columns
                          </a>
                        </code>
                        <code>
                          <a
                            href="http://bulma.io/documentation/columns/basics"
                          >
                            column
                          </a>
                        </code>
                      </td>
                    </tr>
                    <tr>
                      <th>
                        Layout
                      </th>
                      <td>
                        <code>
                          <a
                            href="http://bulma.io/documentation/layout/section"
                          >
                            section
                          </a>
                        </code>
                        <code>
                          <a
                            href="http://bulma.io/documentation/layout/container"
                          >
                            container
                          </a>
                        </code>
                        <code>
                          <a
                            href="http://bulma.io/documentation/layout/footer"
                          >
                            footer
                          </a>
                        </code>
                      </td>
                    </tr>
                    <tr>
                      <th>
                        Elements
                      </th>
                      <td>
                        <code>
                          <a
                            href="http://bulma.io/documentation/elements/button"
                          >
                            button
                          </a>
                        </code>
                        <code>
                          <a
                            href="http://bulma.io/documentation/elements/content"
                          >
                            content
                          </a>
                        </code>
                        <code>
                          <a
                            href="http://bulma.io/documentation/elements/title"
                          >
                            title
                          </a>
                        </code>
                        <code>
                          <a
                            href="http://bulma.io/documentation/elements/title"
                          >
                            subtitle
                          </a>
                        </code>
                      </td>
                    </tr>
                    <tr>
                      <th>
                        Form
                      </th>
                      <td>
                        <code>
                          <a
                            href="http://bulma.io/documentation/form/general"
                          >
                            field
                          </a>
                        </code>
                        <code>
                          <a
                            href="http://bulma.io/documentation/form/general"
                          >
                            control
                          </a>
                        </code>
                      </td>
                    </tr>
                    <tr>
                      <th>
                        Helpers
                      </th>
                      <td>
                        <code>
                          <a
                            href="http://bulma.io/documentation/modifiers/typography-helpers/"
                          >
                            has-text-centered
                          </a>
                        </code>
                        <code>
                          <a
                            href="http://bulma.io/documentation/modifiers/typography-helpers/"
                          >
                            has-text-weight-semibold
                          </a>
                        </code>
                      </td>
                    </tr>
                  </tbody>
                </table>
                <div
                  class="content"
                >
                  <p>
                    If you want to 
                    <strong>
                      learn more
                    </strong>
                    , follow these links:
                  </p>
                </div>
                <div
                  class="field is-grouped"
                >
                  <p
                    class="control"
                  >
                    <a
                      class="button is-medium is-primary"
                      href="http://bulma.io"
                    >
                      <strong
                        class="has-text-weight-semibold"
                      >
                        Bulma homepage
                      </strong>
                    </a>
                  </p>
                  <p
                    class="control"
                  >
                    <a
                      class="button is-medium is-link"
                      href="http://bulma.io/documentation/overview/start/"
                    >
                      <strong
                        class="has-text-weight-semibold"
                      >
                        Documentation
                      </strong>
                    </a>
                  </p>
                </div>
              </div>
            </div>
          </div>
        </section>
        <footer
          class="footer has-text-centered"
        >
          <div
            class="container"
          >
            <div
              class="columns"
            >
              <div
                class="column is-8-desktop is-offset-2-desktop"
              >
                <p>
                  <strong
                    class="has-text-weight-semibold"
                  >
                    <a
                      href="https://www.npmjs.com/package/bulma-start"
                    >
                      bulma-start@0.0.3
                    </a>
                  </strong>
                </p>
                <p>
                  <small>
                     Source code licensed 
                    <a
                      href="http://opensource.org/licenses/mit-license.php"
                    >
                      MIT
                    </a>
                  </small>
                </p>
                <p
                  style="margin-top:1rem;"
                >
                  <a
                    href="http://bulma.io"
                  >
                    <img
                      alt="Made with Bulma"
                      height="24"
                      src="made-with-bulma.png"
                      width="128"
                    />
                  </a>
                </p>
              </div>
            </div>
          </div>
        </footer>
      </div>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "RouterLink": Object {
          "name": "RouterLink",
          "props": Object {
            "activeClass": [Function],
            "ariaCurrentValue": Object {
              "default": "page",
              "type": [Function],
            },
            "custom": [Function],
            "exactActiveClass": [Function],
            "to": Object {
              "required": true,
              "type": Array [
                [Function],
                [Function],
              ],
            },
          },
          "setup": [Function],
        },
        "RouterView": Object {
          "__props": Array [
            Object {
              "name": Object {
                "0": false,
                "1": true,
                "default": "default",
                "type": [Function],
              },
              "route": Object {
                "0": false,
                "1": true,
                "type": [Function],
              },
            },
            Array [
              "name",
            ],
          ],
          "name": "RouterView",
          "props": Object {
            "name": Object {
              "0": false,
              "1": true,
              "default": "default",
              "type": [Function],
            },
            "route": [Function],
          },
          "setup": [Function],
        },
      },
      "config": Object {
        "errorHandler": undefined,
        "globalProperties": Object {
          "$router": Object {
            "addRoute": [Function],
            "afterEach": [Function],
            "back": [Function],
            "beforeEach": [Function],
            "beforeResolve": [Function],
            "currentRoute": Object {
              "__v_isRef": true,
              "value": Object {
                "fullPath": "/",
                "hash": "",
                "matched": Array [],
                "meta": Object {},
                "name": undefined,
                "params": Object {},
                "path": "/",
                "query": Object {},
                "redirectedFrom": undefined,
              },
            },
            "forward": [Function],
            "getRoutes": [Function],
            "go": [Function],
            "hasRoute": [Function],
            "install": [Function],
            "isReady": [Function],
            "onError": [Function],
            "options": Object {
              "history": Object {
                "base": "",
                "createHref": [Function],
                "destroy": [Function],
                "go": [Function],
                "listen": [Function],
                "location": "",
                "push": [Function],
                "replace": [Function],
                "state": Object {},
              },
              "routes": Array [],
            },
            "push": [Function],
            "removeRoute": [Function],
            "replace": [Function],
            "resolve": [Function],
          },
        },
        "isCustomElement": [Function],
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "mixins": Array [
        Object {
          "__props": Array [],
          "beforeCreate": [Function],
        },
        Object {
          "__props": Array [],
          "beforeCreate": [Function],
        },
      ],
      "provides": Object {
        Symbol([vue-router]: router): Object {
          "addRoute": [Function],
          "afterEach": [Function],
          "back": [Function],
          "beforeEach": [Function],
          "beforeResolve": [Function],
          "currentRoute": Object {
            "__v_isRef": true,
            "value": Object {
              "fullPath": "/",
              "hash": "",
              "matched": Array [],
              "meta": Object {},
              "name": undefined,
              "params": Object {},
              "path": "/",
              "query": Object {},
              "redirectedFrom": undefined,
            },
          },
          "forward": [Function],
          "getRoutes": [Function],
          "go": [Function],
          "hasRoute": [Function],
          "install": [Function],
          "isReady": [Function],
          "onError": [Function],
          "options": Object {
            "history": Object {
              "base": "",
              "createHref": [Function],
              "destroy": [Function],
              "go": [Function],
              "listen": [Function],
              "location": "",
              "push": [Function],
              "replace": [Function],
              "state": Object {},
            },
            "routes": Array [],
          },
          "push": [Function],
          "removeRoute": [Function],
          "replace": [Function],
          "resolve": [Function],
        },
        Symbol([vue-router]: route location): Object {
          "fullPath": "/",
          "hash": "",
          "matched": Array [],
          "meta": Object {},
          "name": undefined,
          "params": Object {},
          "path": "/",
          "query": Object {},
          "redirectedFrom": undefined,
        },
      },
      "reload": [Function],
    },
    "_props": null,
    "component": [Function],
    "config": Object {
      "errorHandler": undefined,
      "globalProperties": Object {
        "$router": Object {
          "addRoute": [Function],
          "afterEach": [Function],
          "back": [Function],
          "beforeEach": [Function],
          "beforeResolve": [Function],
          "currentRoute": Object {
            "__v_isRef": true,
            "value": Object {
              "fullPath": "/",
              "hash": "",
              "matched": Array [],
              "meta": Object {},
              "name": undefined,
              "params": Object {},
              "path": "/",
              "query": Object {},
              "redirectedFrom": undefined,
            },
          },
          "forward": [Function],
          "getRoutes": [Function],
          "go": [Function],
          "hasRoute": [Function],
          "install": [Function],
          "isReady": [Function],
          "onError": [Function],
          "options": Object {
            "history": Object {
              "base": "",
              "createHref": [Function],
              "destroy": [Function],
              "go": [Function],
              "listen": [Function],
              "location": "",
              "push": [Function],
              "replace": [Function],
              "state": Object {},
            },
            "routes": Array [],
          },
          "push": [Function],
          "removeRoute": [Function],
          "replace": [Function],
          "resolve": [Function],
        },
      },
      "isCustomElement": [Function],
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.0.0-rc.5",
  },
  "__setProps": [Function],
  "componentVM": Object {
    "__emitted": Object {},
  },
  "rootVM": Object {
    "__emitted": Object {},
  },
}
`;
